src/libs/SidebarUtils.js

```diff
@@ -16,11 +16,17 @@ import * as CollectionUtils from './CollectionUtils';
 // Session also can remain stale because the only way for the current user to change is to sign out and sign in, which would clear out all the Onyx
 // data anyway and cause SidebarLinks to rerender.
 
-let reports;
+const chatReports = {};
+const iouReports = {};
 Onyx.connect({
     key: ONYXKEYS.COLLECTION.REPORT,
-    waitForCollectionCallback: true,
-    callback: val => reports = val,
+    callback: (report, key) => {
+        if (ReportUtils.isIOUReport(report)) {
+            iouReports[key] = report;
+        } else {
+            chatReports[key] = report;
+        }
```

Comment:
NAB: similar to the previous comment, I'm a bit worried about this handling correctly the removal of a report. 

### One option could be to add

```js
if (!report) {
    delete iouReports[key];
    delete chatReports[key];
}
```

### or another option

Keep `waitForCollectionCallback: true,` and changed the body of the callback to something like:

```js
iouReports = _.chain(reports)
    .keys()
    .reduce((memo, key) => {
        if (ReportUtils.isIOUReport(reports[key])) {
            memo[key] = reports[key];
        }
        return memo;
    }, {}).value();

chatReports = _.chain(reports)
    .keys()
    .reduce((memo, key) => {
        if (reports[key] && !ReportUtils.isIOUReport(reports[key])) {
            memo[key] = reports[key];
        }
        return memo;
    }, {}).value();
```

Reply:
Added a report existence check!